godel_surface_detection/src/services/blending_service_path_generation.cpp

```diff
@@ -384,122 +147,188 @@ SurfaceBlendingService::generateProcessPath(const int& id,
-
-  // Marker stuff
-  process_path_results_.process_visualization_.markers.clear();
-  process_path_results_.edge_visualization_.markers.clear();
-  process_path_results_.scan_visualization_.markers.clear();
-  marker_counter_ = 0;
 
   godel_surface_detection::TrajectoryLibrary lib;
-  geometry_msgs::PoseArray blend_poses, edge_poses;
+  // Clear previous results
+  process_path_results_.blend_poses_.clear();
+  process_path_results_.edge_poses_.clear();
+  process_path_results_.scan_poses_.clear();
+
   for (const auto& id : selected_ids)
   {
     // Generate motion plan
-    ProcessPathResult paths = generateProcessPath(id, blend_params, scan_params);
+    ProcessPathResult paths;
+    generateProcessPath(id, paths);
+
+    // If planning failed entirely, skip to next
+    if(paths.paths.size() == 0)
+      continue;
 
-    // Add poses to visualization stack
+    // Add new path to result
     for(const auto& vt: paths.paths)
     {
       if(isBlendingPath(vt.first))
-        blend_poses.poses.insert(blend_poses.poses.end(), vt.second.poses.begin(), vt.second.poses.end());
+        process_path_results_.blend_poses_.push_back(vt.second);
+
       else if(isEdgePath(vt.first))
-        edge_poses.poses.insert(edge_poses.poses.end(), vt.second.poses.begin(), vt.second.poses.end());
+        process_path_results_.edge_poses_.push_back(vt.second);
+
       else if(isScanPath(vt.first))
-        ;
+        process_path_results_.scan_poses_.push_back(vt.second);
+
       else
-        ROS_ERROR_STREAM("Tried to visualize an unrecognized path type: " << vt.first);
+        ROS_ERROR_STREAM("Tried to process an unrecognized path type: " << vt.first);
     }
 
+    ros::NodeHandle nh;
+
+    godel_msgs::BlendingPlanParameters blend_params;
+    blend_params.margin = params.margin;
+    blend_params.overlap = params.overlap;
+    blend_params.tool_radius = params.tool_radius;
+    blend_params.discretization = params.discretization;
+    blend_params.safe_traverse_height = params.traverse_height;
+    nh.getParam(SPINDLE_SPEED_PARAM, blend_params.spindle_speed);
+    nh.getParam(APPROACH_SPD_PARAM, blend_params.approach_spd);
+    nh.getParam(BLENDING_SPD_PARAM, blend_params.blending_spd);
+    nh.getParam(RETRACT_SPD_PARAM, blend_params.retract_spd);
+    nh.getParam(TRAVERSE_SPD_PARAM, blend_params.traverse_spd);
+
+    godel_msgs::ScanPlanParameters scan_params;
+    scan_params.scan_width = params.scan_width;
+    scan_params.margin = params.margin;
+    scan_params.overlap = params.overlap;
+    scan_params.scan_width = params.scan_width;
+    nh.getParam(APPROACH_DISTANCE_PARAM, scan_params.approach_distance);
+    nh.getParam(TRAVERSE_SPD_PARAM, scan_params.traverse_spd);
+    nh.getParam(QUALITY_METRIC_PARAM, scan_params.quality_metric);
+    nh.getParam(WINDOW_WIDTH_PARAM, scan_params.window_width);
+    nh.getParam(MIN_QA_VALUE_PARAM, scan_params.min_qa_value);
+    nh.getParam(MAX_QA_VALUE_PARAM, scan_params.min_qa_value);
```

Comment:
So we get a certain class of parameters from the request message, and the rest are loaded via parameters after each request? I imagine I'll find it in the GUI code, but are these updated when the user selects certain items in the menu?